### ISSUE TITLE
No `severity` value in logs

### ISSUE THREAD

**Issue by schodemeiss** [openobserve/openobserve#3649]
### Which OpenObserve functionalities are the source of the bug?

log search

### Is this a regression?

Yes

### Description

When browsing ingested logs from the Otel Collector via OTLP, the `severity` property is always `null`; even though I can see from the OTEL Collector that the severity number has been set and sent.

It *seems* like something on the way into OpenObserve is dropping that value for some reason?

### Please provide the version you discovered this bug in (check about page for version information)

```true
Version: v0.10.7-rc2
```

**Comment by kanashimia** [comment]
I remember this happening, note that this only applies to a severity number, you can also specify severity string right now, and it should work.

**Comment by schodemeiss** [comment]
Ok, so you're saying the fact it's an integer is the problem, and if I ingested a string that would work?

Is the bug in OpenObserve? Is it something we can fix?

**Comment by kanashimia** [comment]
> Ok, so you're saying the fact it's an integer is the problem, and if I ingested a string that would work?

Yes, if you set `severity_text` in addition or instead of `severity_number`.

> Is the bug in OpenObserve? Is it something we can fix?

I think this is a bug, kinda, in particular it is happening because of this line:
https://github.com/openobserve/openobserve/blob/e5b7476c529428f49703e06d64330b9b57c58e02/src/service/logs/otlp_grpc.rs#L398
As you see it just sets `"severity"` unconditionally depending on the `severity_text`, but the `severity_number` is untouched.

Which was introduced in this commit in which they've added GRPC support: https://github.com/openobserve/openobserve/pull/1399
Note that code between GRPC and HTTP exporters is shared.

It could be set based on the `severity_text` or else `severity_number` converted to a string or else `null`, but then there is a minor problem that you loose access to the severity number, which may be useful for colouring the logs in the future. Alternative is to make them two separate fields, not sure.

CC: @oasisk
